service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java

```diff
@@ -385,12 +392,20 @@ private void publishTempoSamplesToCBioPortal(Set<String> tumorPrimaryIds) throws
                     LOG.error("Invalid Custodian Information for sample with Primary ID " + primaryId);
                     continue;
                 }
+                String cmoPatientId = sampleMetadata.getCmoPatientId();
+                SmilePatient patient = patientService.getPatientByCmoPatientId(cmoPatientId);
                 // build tempo sample object
                 TempoSample tempoSample = TempoSample.newBuilder()
                     .setPrimaryId(primaryId)
                     .setCmoSampleName(cmoSampleName)
                     .setAccessLevel(accessLevel)
                     .setCustodianInformation(custodianInformation)
+                    .setBaitSet(StringUtils.defaultString(sampleMetadata.getBaitSet(), ""))
+                    .setGenePanel(StringUtils.defaultString(sampleMetadata.getGenePanel(), ""))
+                    .setOncotreeCode(StringUtils.defaultString(sampleMetadata.getOncotreeCode(), ""))
+                    .setCmoPatientId(StringUtils.defaultString(cmoPatientId, ""))
+                    .setDmpPatientId(StringUtils.defaultString(patient.getPatientAlias("dmpId"), ""))
```

Comment:
I added `StringUtils.defaultString()` to safely handle instances where these fields are null because when building, `TempoSample` would result in a null pointer exception if it's passed a null value. Technically right now there are no samples of null `genePanel` but I handled it too just in case.